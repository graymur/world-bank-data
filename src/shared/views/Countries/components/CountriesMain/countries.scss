@import "~style/lib";

.countries {
    &__list {
        display: block;

        @include tablet() {
            columns: 2;
            column-gap: 2rem
        }

        @include desktop() {
            columns: 3;
            column-gap: 3rem;
        }

        &__item {
            display: block;
            margin-bottom: .5em;
        }
    }
}