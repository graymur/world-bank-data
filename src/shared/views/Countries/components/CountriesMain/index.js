import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import Helmet from 'react-helmet';
import Loader from 'shared/components/Loader';
import getPageTitle from 'shared/utils/getPageTitle';
import classnames from 'classnames';
import './countries.scss';

export default class CountriesMain extends React.Component {
	static propTypes = {
		loading: PropTypes.bool,
		countries: PropTypes.array
	};

	renderCountries() {
		return (
			<nav className='countries__list'>
				{this.props.countries.map(country => (
					<Link key={country.iso2Code} className='countries__list__item' to={`/countries/${country.iso2Code}`}>{country.name}</Link>
				))}
			</nav>
		);
	}

	render() {
		const {loading} = this.props;
		const classNames = classnames('countries__container', {'loading': loading});

		return (
			<div className='countries'>
				<Helmet>
					<title>{getPageTitle('Countries')}</title>
				</Helmet>
				<h1 className='countries__title'>Countries List</h1>
				<div className={classNames}>
					{loading ? <Loader/> : this.renderCountries()}
				</div>
			</div>
		);
	}
}
