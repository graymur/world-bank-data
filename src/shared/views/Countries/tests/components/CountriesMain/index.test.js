/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import {StaticRouter} from 'react-router';
import CountriesMain from 'shared/views/Countries/components/CountriesMain';

const countries = [{
	iso2Code: 'AA',
	name: 'Country AA'
}, {
	iso2Code: 'BB',
	name: 'Country BB'
}];

test('Shows loading sign', () => {
	const component = renderer.create(
		<StaticRouter location="someLocation" context={{}}>
			<CountriesMain loading={Boolean(true)}/>
		</StaticRouter>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Shows countries list', () => {
	const component = renderer.create(
		<StaticRouter location="someLocation" context={{}}>
			<CountriesMain loading={Boolean(false)} countries={countries}/>
		</StaticRouter>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});