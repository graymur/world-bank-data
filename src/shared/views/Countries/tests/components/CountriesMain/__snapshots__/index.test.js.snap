// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows countries list 1`] = `
<div
  className="countries"
>
  <h1
    className="countries__title"
  >
    Countries List
  </h1>
  <div
    className="countries__container"
  >
    <nav
      className="countries__list"
    >
      <a
        className="countries__list__item"
        href="/countries/AA"
        onClick={[Function]}
      >
        Country AA
      </a>
      <a
        className="countries__list__item"
        href="/countries/BB"
        onClick={[Function]}
      >
        Country BB
      </a>
    </nav>
  </div>
</div>
`;

exports[`Shows loading sign 1`] = `
<div
  className="countries"
>
  <h1
    className="countries__title"
  >
    Countries List
  </h1>
  <div
    className="countries__container loading"
  >
    <span
      className="loader"
      style={
        Object {
          "border": "10px solid #f3f3f3",
          "borderTop": "10px solid #3498db",
          "height": "50px",
          "width": "50px",
        }
      }
    />
  </div>
</div>
`;
